Tests/Common/Securities/SecurityIdentifierTests.cs

```diff
@@ -389,6 +389,347 @@ public void GenerateBase_UsesProvidedSymbol_WhenBaseDataTypeIsNull()
+            Assert.Throws<ArgumentException>(() =>
+            {
+                Symbol.CreateOption(
+                    equity,
+                    Market.USA,
+                    OptionStyle.American,
+                    OptionRight.Call,
+                    999999m,
+                    new DateTime(2020, 5, 21));
+            });
+        }
+
+        [Test]
+        public void HighPrecisionFractionalNonBoundApproachingNumberStrikePricesThrows()
+        {
+            var equity = Symbol.Create("AAPL", SecurityType.Equity, Market.USA);
+            Assert.Throws<ArgumentException>(() =>
+            {
+                Symbol.CreateOption(
+                    equity,
+                    Market.USA,
+                    OptionStyle.American,
+                    OptionRight.Call,
+                    47.5712m, // strike decimal precision is limited to 4 decimal places only
+                    new DateTime(2020, 5, 21));
+            });
+        }
+
+        [Test]
+        public void HighPrecisionNegativeWholeNumberStrikePricesThrows()
+        {
+            var equity = Symbol.Create("AAPL", SecurityType.Equity, Market.USA);
+            Assert.Throws<ArgumentException>(() =>
+            {
+                Symbol.CreateOption(
+                    equity,
+                    Market.USA,
+                    OptionStyle.American,
+                    OptionRight.Call,
+                    -475712m,
+                    new DateTime(2020, 5, 21));
+            });
+        }
+
+        [Test]
+        public void HighPrecisionNegativeFractionalNumberStrikePricesThrows()
+        {
+            var equity = Symbol.Create("AAPL", SecurityType.Equity, Market.USA);
+            Assert.Throws<ArgumentException>(() =>
+            {
+                Symbol.CreateOption(
+                    equity,
+                    Market.USA,
+                    OptionStyle.American,
+                    OptionRight.Call,
+                    -47.5712m, // strike decimal precision is limited to 4 decimal places only
+                    new DateTime(2020, 5, 21));
+            });
+        }
+
+        [Test]
+        public void HighPrecisionNegativeWholeVeryOutOfBoundsNumberStrikePricesThrows()
+        {
+            var equity = Symbol.Create("AAPL", SecurityType.Equity, Market.USA);
+            Assert.Throws<ArgumentException>(() =>
+            {
+                Symbol.CreateOption(
+                    equity,
+                    Market.USA,
+                    OptionStyle.American,
+                    OptionRight.Call,
+                    -999999m,
+                    new DateTime(2020, 5, 21));
+            });
+        }
+
+        [Test]
+        public void HighPrecisionNegativeFractionalNonBoundApproachingNumberStrikePricesThrows()
+        {
+            var equity = Symbol.Create("AAPL", SecurityType.Equity, Market.USA);
+            Assert.Throws<ArgumentException>(() =>
```

Comment:
This test is the same as `HighPrecisionNegativeFractionalNumberStrikePricesThrows`. `HighPrecisionNegativeWholeNumberStrikePricesThrows` seems to cover the same case as `HighPrecisionNegativeWholeVeryOutOfBoundsNumberStrikePricesThrows`. `HighPrecisionFractionalNumberStrikePricesThrows` is the same as `HighPrecisionFractionalNonBoundApproachingNumberStrikePricesThrows`.

Let's create a single test with all these different test cases, we can add a flag to know if we expect an exception. Something like ` [TestCase("-999999", true)]` strike/throws and parse the string to decimal?